pkg/operator/mon/mon.go

```diff
@@ -116,18 +116,41 @@ func (c *Cluster) createMonSecretsAndSave(clientset *kubernetes.Clientset) (*mon
 		return nil, fmt.Errorf("failed to create mon secrets. %+v", err)
 	}
 
-	// store the secrets
+	// store the secrets for internal usage of the rook pods
 	secrets := map[string]string{
 		clusterSecretName: info.Name,
 		fsidSecretName:    info.FSID,
 		monSecretName:     info.MonitorSecret,
 		adminSecretName:   info.AdminSecret,
 	}
-	_, err = clientset.Secrets(c.Namespace).Create(&v1.Secret{ObjectMeta: v1.ObjectMeta{Name: appName}, StringData: secrets})
+	secret := &v1.Secret{
+		ObjectMeta: v1.ObjectMeta{Name: appName}, StringData: secrets,
+		Type: "kubernetes.io/rook",
+	}
+	_, err = clientset.Secrets(c.Namespace).Create(secret)
 	if err != nil {
 		return nil, fmt.Errorf("failed to save mon secrets. %+v", err)
 	}
 
+	// store the secret for usage by the storage class
+	storageClassSecret := map[string]string{
+		"key": info.AdminSecret,
+	}
+	secret = &v1.Secret{
+		ObjectMeta: v1.ObjectMeta{Name: "rook-admin"},
+		StringData: storageClassSecret,
+		Type:       "kubernetes.io/rbd",
+	}
+	_, err = clientset.Secrets(c.Namespace).Create(secret)
+	if err != nil {
+		if !k8sutil.IsKubernetesResourceAlreadyExistError(err) {
+			return nil, fmt.Errorf("failed to save rbd consumer secret. %+v", err)
+		}
+		logger.Infof("rook-admin secret already exists")
```

Comment:
we call it `rbd consumer secret` in the error message, but `rook-admin secret` in this log and the next log statement